### ISSUE TITLE
Parsing error on private attribute (NodeJS 12)

### ISSUE THREAD

**Issue by neolao** [eslint/eslint#11820]
**Tell us about your environment**

* **ESLint Version:** 5.16.0
* **Node Version:** 12.4.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

`hello.js`:
```js
class Hello {
  #foo = "bar";
}
```

```bash
$ eslint ./hello.js
```

**What did you expect to happen?**

No parsing error

**What actually happened? Please include the actual, raw output from ESLint.**
```
/home/me/hello.js
  2:3  error  Parsing error: Unexpected character '#'

✖ 1 problem (1 error, 0 warnings)
```

**Are you willing to submit a pull request to fix this bug?**

**Comment by kaicataldo** [comment]
Private class field declarations are currently [stage 3](https://github.com/tc39/proposal-class-fields/blob/master/README.md). ESLint’s core rules and default parser will be updated to support this syntax if and when it reaches stage 4. In the meantime, you can use another parser like babel-eslint to lint experimental syntax.
